### PR TITLE
Adding Some Methods

### PR DESCRIPTION
**Pull request by RuanLucasGD**
Hello, in this PR i added some methods:

- Vector2.Normalized
- Float2.Normalized
- Gamepad.IsAnyKeyDown
- Mouse.IsAnyKeyDown

### CODE REVIEW

File: Source/Engine/Core/Math/Vector2.cs [hunk 1]
```diff
@@ -301,6 +301,19 @@ public void Normalize()
...
+        /// <summary>
+        /// Gets the normalized vector. Returned vector has length equal 1.
+        /// </summary>
+        public Vector2 Normalized
+        {
+            get
+            {
+                Vector2 result = this;
+                result.Normalize();
+                return result;
```
Comment (Tryibion) [inline Source/Engine/Core/Math/Vector2.cs:313]: could this just be simplified to `return this.Normalize()`?

**Comment by Tryibion** [comment]
Is there already a function for Keyboard any key down? could that be easy to add as well if it doesnt exist?

**Comment by RuanLucasGD** [comment]
@Tryibion This function already exists, but the ones I added didn't exist

File: Source/Engine/Core/Math/Vector2.cs (hunk 1 above)
Comment (RuanLucasGD) [reply Source/Engine/Core/Math/Vector2.cs:313]: That would simplify things, thanks I'll adjust

Comment (RuanLucasGD) [reply Source/Engine/Core/Math/Vector2.cs:313]: No,` Normalize()` returns a void

Comment (Tryibion) [reply Source/Engine/Core/Math/Vector2.cs:313]: Ah, okay didn't know that. Makes sense then. Could something like this also be useful for vector3?

Comment (RuanLucasGD) [reply Source/Engine/Core/Math/Vector2.cs:313]: It already exists, but I found it strange that the same function did not exist in Vector2 so I created it. I just copied the same function from Vector3 to Vector2

**Comment by mafiesto4** [comment]
I think it would be better to name new methods `IsAnyButtonDown` for Gamepad and Mouse.

**Comment by RuanLucasGD** [comment]
@mafiesto4 IsAnyKeyDown exists in the Keyboard class but does not exist in the Mouse and Gamepad class. What I did was create a copy of what was on the keyboard for the gamepad/mouse.

So I think that if you want to change the mouse/gamepad IsAnyKeyDown, you should also change the keyboard

https://github.com/FlaxEngine/FlaxEngine/blob/997baa3b774ac4dea739e15a55070519b0863afc/Source/Engine/Input/Keyboard.h#L70

**Comment by RuanLucasGD** [comment]
@mafiesto4 What do you think of IsAnyKeyDown for keyboard and IsAnyButtonDown for gamepad/mouse

**Comment by PrecisionRender** [comment]
> @mafiesto4 What do you think of IsAnyKeyDown for keyboard and IsAnyButtonDown for gamepad/mouse?

Personally, I think this is the best way to go.

**Comment by RuanLucasGD** [comment]
@mafiesto4 done

**Pull request merged**